overlord/snapstate/component.go

```diff
@@ -30,8 +31,189 @@ import (
 	"github.com/snapcore/snapd/overlord/state"
 	"github.com/snapcore/snapd/snap"
 	"github.com/snapcore/snapd/snap/naming"
+	"github.com/snapcore/snapd/store"
+	"github.com/snapcore/snapd/strutil"
 )
 
+// InstallComponents installs all of the components in the given names list. The
+// snap represented by info must already be installed, and all of the components
+// in names should not be installed prior to calling this function.
+//
+// TODO:COMPS: respect the transaction that is passed to this function
+func InstallComponents(ctx context.Context, st *state.State, names []string, info *snap.Info, opts Options) ([]*state.TaskSet, error) {
```

Comment:
should info come before names? what are doing order wise typically?

Reply:
In the preexisting `InstallComponentPath`, the identifier for the component (a `snap.ComponentSideInfo` comes before the `snap.Info`. I figured this follows a similar pattern.